### PR TITLE
Add typescript support

### PR DESCRIPTION
**Pull request by ipfans**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
None.

##### Description of change
Add `index.d.ts` for typescript support.

### CODE REVIEW

**Comment by fengmk2** [comment]
@whxaxes Please review it.

**Comment by whxaxes** [comment]
@ipfans Add an unittest for ts ?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Redis, RedisOptions } from "ioredis";
+
+interface RedisOpts extends RedisOptions {
```
Comment (popomore) [inline index.d.ts:3]: 可以叫 EggRedisOptions

Comment (ipfans) [reply index.d.ts:3]: @popomore 下面另外一个已经用掉了

**Comment by ipfans** [comment]
@whxaxes sorry, I doesn't have any experiences about this. Any instructions about this?

**Comment by whxaxes** [comment]
@ipfans  hi, see https://github.com/eggjs/egg/blob/master/test/ts/index.test.js , create a ts fixture, and use `egg-bin --ts` or `tsc ` to make sure the `d.ts` can be parse correctly.

**Comment by whxaxes** [comment]
@ipfans  hi, any progress ?

**Comment by qiqizjl** [comment]
Is there any progress in this matter? Every time you need to update the node_module file, this is an unpleasant thing. I very much hope to support TS.

**Comment by whxaxes** [comment]
https://github.com/eggjs/egg-redis/pull/23/files  has merged, so close this

**Pull request closed without merging**